### PR TITLE
Trash & color Palette

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Added Trash & Color Palette Functionality.
- Added Trash note using API.
- Added color Selector Feature for Notes.
- Can change & edit color and text.

### CODE REVIEW

**Comment by Jay-Gadhiya** [comment]
you can use model in color palate opening and closing.

**Comment by Jay-Gadhiya** [comment]
image

You can make in row for batter reading notes.

File: Gravity-Notes/src/Components/Notes-card/Notecard.js [hunk 1]
```diff
@@ -9,40 +9,68 @@ import { useAuth } from "../../Contexts/Auth-context"
...
+       <button onClick={()=>applyColorOnCard("orange",note)} className="card-color-btn orange"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("yellow",note)} className="card-color-btn yellow"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("green",note)} className="card-color-btn green"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("purple",note)} className="card-color-btn purple"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("pink",note)} className="card-color-btn pink"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("brown",note)} className="card-color-btn brown"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+     </div> }
+     </div>
```
Comment (Jay-Gadhiya) [inline Gravity-Notes/src/Components/Notes-card/Notecard.js:74]: you can make array of color name and return buttons.

**Comment by Jay-Gadhiya** [comment]
- context and reducer coded well
- I like your input element interface in UI.

**Review by Jay-Gadhiya — APPROVED** [review]
good work @jeetbhalani15

File: Gravity-Notes/src/Components/Notes-card/Notecard.js (hunk 1 above)
Comment (jeetbhalani15) [reply Gravity-Notes/src/Components/Notes-card/Notecard.js:74]: oh ok thanks for this suggestion, @Jay-Gadhiya  and I will refactor my CSS Soon first going with adding all functionality first

Comment (SHIKHASINGH1506) [reply Gravity-Notes/src/Components/Notes-card/Notecard.js:74]: You can create a separate component for color pallete. Just a suggestion.

File: Gravity-Notes/src/Components/Notes-card/Notecard.css [hunk 2]
```diff
@@ -24,7 +24,7 @@
     word-wrap: break-word;
     padding: 1rem;
 }
-.note button{
+.card-action-btn{
     position: relative;
     float: right;
     color: #3B3B3B;
```
Comment (SHIKHASINGH1506) [inline Gravity-Notes/src/Components/Notes-card/Notecard.css:30]: use variables for color

File: Gravity-Notes/src/Components/Notes-card/Notecard.js [hunk 3]
```diff
@@ -9,40 +9,68 @@ import { useAuth } from "../../Contexts/Auth-context"
...
 
     const editHandler = async (note)=>{
          setNote((pre =>({...pre,
             title: note.title,
             content: note.content,
-            id: note._id,
+            _id: note._id,
             flag: true,
+            color:note.color,
             date: new Date(Date.now()).toLocaleString().split(','[0])
```
Comment (SHIKHASINGH1506) [inline Gravity-Notes/src/Components/Notes-card/Notecard.js:30]: You can create a util function for getting current date. That will be reusable and you don't have to write this "new Date(Date.now()).toLocaleString().split(','[0])" everywhere

File: Gravity-Notes/src/Components/TextArea.js/Textarea.js [hunk 4]
```diff
@@ -9,17 +9,29 @@ import axios from "axios";
...
+import { BiTagAlt } from "react-icons/bi";
 
 export const Textarea = () => {
   const {authState} = useAuth();
   const {noteState, noteDispatch,note,setNote} = useNotes();
   const [isExpanded, setisExpanded] = useState(false);
+  const [colorSelector, setColorSelector] = useState(false);
 
   const inputHandler = (e)=>{
       setNote(pre=>({...pre,[e.target.name]:e.target.value, data: new Date(Date.now()).toLocaleString().split(','[0])}))
```
Comment (SHIKHASINGH1506) [inline Gravity-Notes/src/Components/TextArea.js/Textarea.js:23]: can use that date util function here also. Code will look more clean

**Review by SHIKHASINGH1506 — APPROVED** [review]
Left few suggestions. 
You can create one note card and use it for archive also, as react is all about re usability. That goes in refactoring your code once you get enough time. 
I liked your logic, it's very clear.

**Pull request merged**